sbi/neural_nets/factory.py

```diff
@@ -266,17 +293,37 @@ def build_fn_snpe_a(batch_theta, batch_x, num_components):
 
     def build_fn(batch_theta, batch_x):
         if model == "mdn":
-            return build_mdn(batch_x=batch_theta, batch_y=batch_x, **kwargs)
+            return build_mdn(batch_x=batch_x, batch_y=batch_theta, **kwargs)
```

Comment:
why have you replaced the order of `batch_x` and `batch_theta` here?

Reply:
It was an automatic fix of ruff and I overlooked it after correction.